From b2ed72bc643a251cbcdca7a0d1058517824f3aa9 Mon Sep 17 00:00:00 2001
From: Thilo Graf <dbt@novatux.de>
Date: Wed, 18 Nov 2015 19:43:06 +0100
Subject: [PATCH 10/21] CAudioPlayerGui:try to fix infocklock view in
 audioplayer.

after closed screensaver, infoclock was painted. That looks not good.
This should fix this.
---
 src/gui/audioplayer.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/gui/audioplayer.cpp b/src/gui/audioplayer.cpp
index 23d8bdc..4d1e9cc 100644
--- a/src/gui/audioplayer.cpp
+++ b/src/gui/audioplayer.cpp
@@ -69,7 +69,7 @@
 #include <gui/screensaver.h>
 #include "gui/pictureviewer.h"
 extern CPictureViewer * g_PicViewer;
-
+#include <gui/infoclock.h>
 #include <system/settings.h>
 #include <system/helpers.h>
 #include <driver/screen_max.h>
@@ -334,7 +334,8 @@ int CAudioPlayerGui::show()
 
 	// clear whole screen
 	m_frameBuffer->paintBackground();
-
+	CInfoClock::getInstance()->block();
+	CScreenSaver::getInstance()->OnAfterStop.connect(sigc::mem_fun(CInfoClock::getInstance(), &CInfoClock::block));
 	CVFD::getInstance()->setMode(CVFD::MODE_AUDIO);
 	paintLCD();
 
@@ -848,7 +849,8 @@ int CAudioPlayerGui::show()
 
 	if (m_state != CAudioPlayerGui::STOP)
 		stop();
-
+	CInfoClock::getInstance()->enableInfoClock(CInfoClock::getInstance()->isRun());
+	CScreenSaver::getInstance()->OnAfterStop.clear();
 	return ret;
 }
 
-- 
2.1.4

